### ISSUE TITLE
Add automatic processing for Montreal data

### ISSUE THREAD

**Issue by mschoettle** [jeremymoreau/covid19mtl#40]
`data_mtl.csv` (history of data for Montreal) is updated manually currently. This is done in the Excel spreadsheet and then exported from there.

The data needs to be pulled from several places (e.g., `data_mtl_new_cases` for new/cumulative cases, `data_mtl_age.csv` for age distribution) plus some calculations.

I am wondering whether this needs to be in a single file or if it would be possible to separate it into several files depending on the data (e.g., one for the case history in MTL, one for age distribution etc.).

**Comment by jeremymoreau** [comment]
I don't think there is really any reason it needs to be a single file, we could split it into separate files if you feel that is more logical. We'd need to make some small changes to `core.py`, but that should be very straightforward.

**Comment by mschoettle** [comment]
It is more for processing I think. Might be easier to separate some files instead of combining data from several files.

**Comment by mschoettle** [comment]
So I think there are two parts (maybe even three):

1. new cases + cumulative cases
2. new deaths + cumulative deaths
3. age distribution of cases (+ per 100000)

The data for 1. can be found in `data_mtl_new_cases.csv` but it only contains that. 

```python
df = pd.read_csv('app/data/sources/2020-11-30/data_mtl_new_cases.csv', sep=';')
# drop empty last rows
df_cleaned = df[df['Date'].notna()]
# drop empty last row
df_cleaned.iloc[:, 0:-1]
```

`data_qc` contains information for Montreal for 1. and 2. and the CSV contains the whole history so it is much easier to process.

The numbers between the data coming from Santé Montreal and INSPQ are slightly different actually, probably because of the exact time of the day they were extracted.

For 3. there is `data_mtl_age.csv`. Unfortunately the numbers don't match between that file and `data_mtl.csv` (checked for 2020-06-01) but starting from June 1st (when the age distribution file is available) we could use `data_mtl_age.csv` to build the rest of dates on top of the existing `data_mtl.csv`.

**Comment by jeremymoreau** [comment]
That sounds reasonable to me. I think the mismatch might in part be due to Santé Montreal updating past data (the data in `data_mtl.csv` was recorded only once the day of initial posting). Also the population numbers I used might be slightly different (Santé Montreal did not initially report cases per 100,000 for age groups).


age group | population
-- | --
0-4 | 109740
5-9 | 104385
10-19 | 188185
20-29 | 293225
30-39 | 299675
40-49 | 254475
50-59 | 258875
60-69 | 205005
70-79 | 129680
80+ | 98805

**Comment by mschoettle** [comment]
It is easier to use `data_qc.csv` but just to note how to process `mtl_new_cases.csv`:

```python
# replace column names, ignore empty last column
cases_df = pd.read_csv(
    cases_csv,
    sep=';',
    header=0,
    names=['date', 'new_cases', 'total_cases'],
    usecols=[0, 1, 2],
    encoding='utf-8'
)

# remove rows with all NaN (there are some at the end)
cases_df.dropna(how='all', inplace=True)
```
